tests/cpp/unit-tests/test_configopts.cpp

```diff
@@ -0,0 +1,293 @@
+/*
+ * BSD 2-Clause License
+ *
+ * Copyright (c) 2021-2023, Hewlett Packard Enterprise
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "../../../third-party/catch/single_include/catch2/catch.hpp"
+#include "../client_test_utils.h"
+#include "logger.h"
+#include "configoptions.h"
+
+unsigned long get_time_offset();
+
+using namespace SmartRedis;
+
+SCENARIO("Testing for ConfigOptions", "[CfgOpts]")
```

Comment:
I think the only other test that I could think of is a larger integration test. While it's beyond the scope to add every configurable variable to this PR, maybe you could add `CLUSTERED`, `SSDB`, and maybe `SR_THREADS` to test that the 'default', 'from environment', and the override methods actually yield a `ConfigOptions` object that has the expected variables.

Reply:
I intentionally stuck with options that won't be set elsewhere so that the test can have full control of the environment. We could definitely add a test like this but it would be a bit tricky to make sure we're in the environment we expect and account for that.

Reply:
I'm going to say that such a test is out of scope for now though

Reply:
I'm ok with that